Sukanta_Roy/progress.md

```diff
@@ -3,78 +3,103 @@
 **Project: Questions+**
 
 Overall Direction:
-- 
-- 
-- 
-
+- MCQ questions:
+	- Attach weights to options
+	- Calculate more statistics (Total, average)
+	- Ensure downloaded data contains all data shown in the page.
+- MSQ questions:
+	- Attach weights to options
+	- Calculate more statistics
+	- Ensure downloaded data contains all data shown in the page.
+- Rubric questions:
+	- Attach weights to each cell instead of each column
+	- Update results accordingly
+	- Ensure downloaded data contains all data shown in the page.
+- Add a new question template
 ---
 
 Week 1 Plan:
-- [ ] 
-- [ ] 
-- [ ] 
+- [ ] Work on [Instructor: edit rubric question: tweak UI to prevent misunderstanding of weights](https://github.com/TEAMMATES/teammates/issues/8824)
+- [ ] Implement the backend part of [MCQ questions: support attaching weights to options](https://github.com/TEAMMATES/teammates/issues/2776)
+	- [ ] Update `QuestionDetails` class
+	- [ ] Update `Logic` class
+	- [ ] Update `Action` class
+	- [ ] Add Corrosponding Unit and Integration tests
 
 Week 2 Plan:
-- [ ] 
-- [ ] 
-- [ ] 
+**Final Semester Exams**
+- [ ] Finalize the sample question list for the new question template
 
 Week 3 Plan:
-- [ ] 
-- [ ] 
-- [ ] 
+- [ ] Finish off remaining work from week 1 (if any)
+- [ ] Implement the frontend part of [MCQ questions: support attaching weights to options](https://github.com/TEAMMATES/teammates/issues/2776)
+	- [ ] Update question template to add the new UI for MCQ weights
+	- [ ] Add JS functionalities
+	- [ ] Add necessary JS tests
+	- [ ] Add frontend UI tests
+- [ ] If frontend and backend part for MCQ weights are finished, then add necessary UI tests and make the PR ready for review.
 
 Week 4 Plan:
-- [ ] 
-- [ ] 
-- [ ] 
+- [ ] Finish off remaining work from past week (if any)
+- [ ] Update documentation for MCQ questions in the `InstructorHelp` page
+- [ ] Code review and fix
```

Comment:
Code review should be progress based. So after u finish the backend part (after week 1, maybe Friday, weekend or something), we will have a code review for your backend implementation. Same applied for frontend, documentation etc.

Reply:
@xpdavid Then, should I update the plan accordingly or should I leave it for now and I can update it later (while adding the progress report) if need be?

Reply:
It's up to you. After all, I will chase u everyday. >_< :D